[v1.1.2]Add network relation explanation (#291)
harvester/docs: docs/networking/clusternetwork.md

Network configuration only works under a certain cluster network. Each network configuration corresponds to a set of hosts with uniform network specifications. Therefore, multiple network configurations are required for a cluster network on non-uniform hosts.

### VM Network

A VM network is an interface in a virtual machine that connects to the host network. As with network configuration, every network except the built-in [management network](./harvester-network.md#management-network) must be under a cluster network.

Harvester supports adding multiple networks to one VM. If a network's cluster network is not enabled on some hosts, the VM that owns this network will not be scheduled to those hosts.

Please refer to [network part](./harvester-network.md) for more details about networks.

### Relation among Cluster Network, Network Configuration, and VM Network
The diagram below shows the relation among cluster network, network configuration, and VM network.

![](/img/v1.2/networking/relation.png)

- Every network config and VM network is under a cluster network.

> - You can categorize hosts based on network specifications by assigning a label to each host, and then add a network config for each group of hosts through a node selector. For example, in the diagram above, for ClusterNetwork-A, we group all hosts into 3 groups, respectively:

Reviewer: ```suggestion
- To categorize hosts based on network specifications, you can assign a label to each host and then add a network config for each group of hosts using a node selector. For ClusterNetwork-A in the diagram above, we group all hosts into 3 groups, respectively:
```

>    - The first group includes host0 and host2, corresponding to network-config-A

Author: No, there is ambiguity in the diagram. Let me clear it.

>    - The first group includes host0 and host2, corresponding to network-config-A

Author: The `network-config-A` contains `host0` and `host2`, and `network-config-D` contains `host2` and `host5`.

>    - The first group includes host0 and host2, corresponding to network-config-A

Reviewer 2: Thanks, the figure looks clear now.

> - Every network config and VM network is under a cluster network.
> - To categorize hosts based on network specifications, you can assign a label to each host and then add a network config for each group of hosts using a node selector. For ClusterNetwork-A in the diagram above, we group all hosts into 3 groups, respectively:
>    - The first group includes host0 and host2, corresponding to network-config-A
>    - The second group includes host1, corresponding to network-config-B
>    - The third group includes the remaining host3, host4 and host5, corresponding to no network config.

Reviewer 3: ```suggestion
All `Network Configs` and `VM Networks` are grouped under a cluster network. To categorize hosts based on their network specifications, a label can be assigned to each host and a network config can be added for each group of hosts using a node selector. E.g., In the diagram above,  for `ClusterNetwork-A` i, hosts are divided into three groups as follows:
   - The first group includes host0 and host2, which correspond to `network-config-A`.
   - The second group includes host1, which corresponds to `network-config-B`.
   - The third group includes the remaining hosts (host3, host4, and host5), which do not have any related network config and therefore do not belong to `ClusterNetwork-A`.

```

>    - The third group includes the remaining host3, host4 and host5, corresponding to no network config.

Reviewer 3: the description that u added here does not match the diagram, can u TPAL?

>    - The first group including host0 and host1, corresponding to network-config-A
>    - The second group including host2, corresponding to network-config-B
>    - The third group include remaining host3, host4 and host5, corresponding to no network config.

Reviewer: ```suggestion
   - The first group includes host0 and host1, corresponding to network-config-A
   - The second group includes host2, corresponding to network-config-B
   - The third group includes the remaining host3, host4 and host5, corresponding to no network config.
```

- The cluster network will only be effective on hosts that are covered by the network configuration. If a VM uses a VM network under a specific cluster network, it can only be scheduled on a host where the cluster network is active.
   In the diagram above, we can see that:
   - ClusterNetwork-A is active on host0, host1 and host2. VM0 uses VM-network-A so it can be scheduled on host0, host1 or host2.

>    - VM1 which uses both VM-network-A and VM-network-B can be only scheduled on host2 where both ClusterNetwork-A and ClusterNetwork-B are active.

Reviewer: ```suggestion
   - VM1, which uses both VM-network-A and VM-network-B, can be only scheduled on host2 where both ClusterNetwork-A and ClusterNetwork-B are active.
```

> - The cluster network will only be effective on hosts that are covered by the network configuration. If a VM uses a VM network under a specific cluster network, it can only be scheduled on a host where the cluster network is active.
>    In the diagram above, we can see that:
>    - ClusterNetwork-A is active on host0, host1 and host2. VM0 uses VM-network-A so it can be scheduled on host0, host1 or host2.
>    - VM1, which uses both VM-network-B and VM-network-C, can be only scheduled on host2 where both ClusterNetwork-A and ClusterNetwork-B are active.
>    - VM0, VM1 and VM2 cannot run on host3 where the two cluster networks are inactive.

Reviewer 3: ```suggestion
The cluster network is only effective on hosts that are covered by the network configuration. If a VM uses a `VM network` under a specific cluster network, it can only be scheduled on a host where the cluster network is active. In the diagram, we can see that:
   - `ClusterNetwork-A` is active on host0, host1, and host2. `VM0` uses `VM-network-A`, so it can be scheduled on any of these hosts.
   - `VM1` uses both `VM-network-B` and `VM-network-C`, so it can only be scheduled on host2 where both `ClusterNetwork-A` and `ClusterNetwork-B` are active.
   - `VM0`, `VM1`, and `VM2` cannot run on host3 where the two cluster networks are inactive.

Overall, this diagram provides a clear visualization of the relationship between cluster networks, network configurations, and VM networks, as well as how they impact VM scheduling on hosts.
```

>    - VM0, VM1 and VM2 cannot run on host3 where the two cluster networks are inactive.

Reviewer 3: ditto

Network configuration only works under a certain cluster network. Each network configuration corresponds to a set of hosts with uniform network specifications. Therefore, multiple network configurations are required for a cluster network on non-uniform hosts.

### VM Network

> A VM network is an interface in a virtual machine that connects to the host network. As with network configuration, every network except the built-in [management network](./harvester-network.md#management-network) must be under a cluster network.

Reviewer 4: ```suggestion
A VM network is an interface in a virtual machine that connects to the host network. As with a network configuration, every network except the built-in [management network](./harvester-network.md#management-network) must be under a cluster network.
```

Harvester supports adding multiple networks to one VM. If a network's cluster network is not enabled on some hosts, the VM that owns this network will not be scheduled to those hosts.

Please refer to [network part](./harvester-network.md) for more details about networks.

### Relationship Between Cluster Network, Network Config, VM Network

> The following diagram shows the relationship between cluster network, network config, and VM network.

Reviewer 4: ```suggestion
The following diagram shows the relationship between a cluster network, a network config, and a VM network.
```

![](/img/v1.2/networking/relation.png)

> All `Network Configs` and `VM Networks` are grouped under a cluster network. To categorize hosts based on their network specifications, a label can be assigned to each host and a network config can be added for each group of hosts using a node selector. E.g., In the diagram above,  for `ClusterNetwork-A` i, hosts are divided into three groups as follows:

Reviewer 4: ```suggestion
All `Network Configs` and `VM Networks` are grouped under a cluster network. 

- A label can be assigned to each host to categorize hosts based on their network specifications.  
- A network config can be added for each group of hosts using a node selector. 

For example, in the diagram above, the hosts in `ClusterNetwork-A` are divided into three groups as follows:
```

>    - The first group includes host0 and host2, which correspond to `network-config-A`.

Reviewer 4: ```suggestion
- The first group includes host0 and host2, which correspond to `network-config-A`.
```

   - The second group includes host1, which corresponds to `network-config-B`.

>    - The third group includes the remaining hosts (host3, host4, and host5), which do not have any related network config and therefore do not belong to `ClusterNetwork-A`.

Reviewer 4: ```suggestion
- The third group includes the remaining hosts (host3, host4, and host5), which do not have any related network config and therefore do not belong to `ClusterNetwork-A`.
```

> The cluster network is only effective on hosts that are covered by the network configuration. If a VM uses a `VM network` under a specific cluster network, it can only be scheduled on a host where the cluster network is active. In the diagram, we can see that:

Reviewer 4: ```suggestion
The cluster network is only effective on hosts that are covered by the network configuration. A VM using a `VM network` under a specific cluster network can only be scheduled on a host where the cluster network is active.

 In the diagram above, we can see that:
```

>    - `ClusterNetwork-A` is active on host0, host1, and host2. `VM0` uses `VM-network-A`, so it can be scheduled on any of these hosts.

Reviewer 4: ```suggestion
- `ClusterNetwork-A` is active on host0, host1, and host2. `VM0` uses `VM-network-A`, so it can be scheduled on any of these hosts.
```

>    - `VM1` uses both `VM-network-B` and `VM-network-C`, so it can only be scheduled on host2 where both `ClusterNetwork-A` and `ClusterNetwork-B` are active.

Reviewer 4: ```suggestion
- `VM1` uses both `VM-network-B` and `VM-network-C`, so it can only be scheduled on host2 where both `ClusterNetwork-A` and `ClusterNetwork-B` are active.
```

>    - `VM0`, `VM1`, and `VM2` cannot run on host3 where the two cluster networks are inactive.

Reviewer 4: ```suggestion
- `VM0`, `VM1`, and `VM2` cannot run on host3 where the two cluster networks are inactive.
```

- The text `enp7s3 (1/3 Down)` in the NICs drop-down list indicates that the enp7s3 NIC is down in one of the three selected nodes. In this case, you need to find the NIC, set it up, and refresh this page. After this, it should be selectable.

:::

:::note

> - Harvester supports updating and deleting network configs. You have to stop affected VMs before action execution.

Reviewer 3: ```suggestion
- Starting from Harvester v1.1.2, Harvester supports updating network configs, but it is required to stop all affected VMs first.
```

- The text `enp7s3 (1/3 Down)` in the NICs drop-down list indicates that the enp7s3 NIC is down in one of the three selected nodes. In this case, you need to find the NIC, set it up, and refresh this page. After this, it should be selectable.

:::

:::note

> - Starting from Harvester v1.1.2, Harvester supports updating network configs, but it is required to stop all affected VMs first.

Reviewer 4: ```suggestion
Starting with Harvester v1.1.2, Harvester supports updating network configs. Make sure to stop all affected VMs before updating network configs.
```

- The text `enp7s3 (1/3 Down)` in the NICs drop-down list indicates that the enp7s3 NIC is down in one of the three selected nodes. In this case, you need to find the NIC, set it up, and refresh this page. After this, it should be selectable.

:::

:::note

> Starting with Harvester v1.1.2, Harvester supports updating network configs. Make sure to stop all affected VMs before updating network configs.

Reviewer 3: @yaocw2020, we may need to mention the upgrade issue of https://github.com/harvester/harvester/issues/3223 here and say that it will get addressed in v1.1.2. WDUT? Thanks.

Author: Say it will get addressed in v1.1.2 in the issue or document?
